drivers/entropy/entropy_stm32.c

```diff
@@ -355,59 +504,22 @@ static int entropy_stm32_rng_get_entropy_isr(const struct device *dev,
 	if (len) {
 		unsigned int key;
 		int irq_enabled;
+		bool rng_already_acquired;
 
 		key = irq_lock();
 		irq_enabled = irq_is_enabled(IRQN);
 		irq_disable(IRQN);
 		irq_unlock(key);
 
-		/* do not proceed if a Seed error occurred */
-		if (LL_RNG_IsActiveFlag_SECS(entropy_stm32_rng_data.rng) ||
-			LL_RNG_IsActiveFlag_SEIS(entropy_stm32_rng_data.rng)) {
+		rng_already_acquired = z_stm32_hsem_is_locked(CFG_HW_RNG_SEMID);
+		acquire_rng();
```

Comment:
Thanks for the review! Concerning the `acquire_rng()` in ISR, I've considered that not to be a issue in this case since this is done only when `entropy_stm32_rng_get_entropy_isr` is called with `ENTROPY_BUSYWAIT`. When this flag is specified, entropy will be generated from ISR, which can take time, so the user calling this routine should be aware that the routine might block. It seems to be used in Zephyr to generate entropy at early boot where it may be desirable to wait for the semaphore. Do you think the routine should instead return an error in case the RNG is not available immediately?
https://github.com/zephyrproject-rtos/zephyr/blob/3702f7220affc5bcf377c4e6c41fc320423d5045/kernel/init.c#L378

Concerning `z_stm32_hsem_is_locked`, I made an error: I've wrongly considered `LL_HSEM_IsSemaphoreLocked` would return true iff the semaphore was currently owned by the core the code is running on. I've just fixed that. No critical section should be required here because we are in ISR so the RNG cannot be acquired by this core (from `start_pool_filling`) between the moment `rng_already_acquired` is set and `acquire_rng` is called, and if the semaphore was owned by this core when `rng_already_acquired` is set, it won't be released concurrently since RNG IRQ is disabled.

Reply:
Right, I missed the test on ENTROPY_BUSYWAIT, so it is not an issue to block on semaphore.

And now I better understand what you intend to do with z_stm32_hsem_is_owned